stablehlo/reference/Tensor.h

```diff
@@ -107,6 +122,53 @@ inline raw_ostream &operator<<(raw_ostream &os, Tensor tensor) {
   return os;
 }
 
+/// Iterator over the indices of a tensor in  major-to-minor order. As an
+/// example, for a tensor with shape [2,3], the iterator enumerates the
+/// following indices (0,0), (0,1), (0,2), (1,0), (1,1), and (1,2) in that
+/// order.
+class DimensionIterator {
```

Comment:
Also, `DimensionIterator` looks sufficiently independent from `Tensor`, so I'd move it into a dedicated target, e.g. `Index.h` and `Index.cpp`.

Reply:
done